internal/apiserver/root.go

```diff
@@ -498,6 +507,57 @@ func setNamespaceOperatingMode() error {
 	return nil
 }
 
+// setRandomPgouserPasswords looks through the pgouser secrets in the Operator's
+// namespace. If any have an empty password, it generates a random password,
+// Base64 encodes it, then stores it in the relevant PGO user's secret
+func setRandomPgouserPasswords() {
+	ctx := context.TODO()
+
+	selector := "pgo-pgouser=true,vendor=crunchydata"
+	secrets, err := Clientset.CoreV1().Secrets(PgoNamespace).
+		List(ctx, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		log.Errorf("Could not get pgouser secrets in namespace: %s", PgoNamespace)
+	}
+
+	for _, secret := range secrets.Items {
+		currentPassword := string(secret.Data["password"])
+
+		// check if password is empty
+		if currentPassword == "" {
+			log.Infof("Password in pgouser secret %s for operator installation %s in namespace %s is empty. "+
+				"Setting a generated password.", secret.Name, InstallationName, PgoNamespace)
+
+			// generate the password using the default password length
+			generatedPassword, err := util.GeneratePassword(util.DefaultGeneratedPasswordLength)
+
+			if err != nil {
+				log.Errorf("Could not generate password for pgouser secret %s for operator installation %s in "+
+					"namespace %s", secret.Name, InstallationName, PgoNamespace)
+			}
+
+			// create the password patch
+			patch, err := json.Marshal(map[string]interface{}{
+				"data": map[string]interface{}{
+					"password": base64.StdEncoding.EncodeToString([]byte(generatedPassword)),
+				},
+			})
```

Comment:
We recently added a type for this in `kubeapi`:

```go
patch, err := kubeapi.NewMergePatch().
	Add("data", "password")(base64.StdEncoding.EncodeToString([]byte(generatedPassword))).
	Bytes()
```

You can also skip the base64 on our end by passing it through `stringData` instead:

```suggestion
			patch, err := kubeapi.NewMergePatch().
				Add("stringData", "password")(generatedPassword).
				Bytes()
```

Reply:
ok, great. I'll make that update. Thanks